modules/core/src/main/java/org/apache/ignite/internal/processors/cache/verify/VerifyBackupPartitionsTaskV2.java

```diff
@@ -538,29 +504,31 @@ private boolean isCacheMatchFilter(DynamicCacheDescriptor desc) {
         /**
          * @param grpCtx Group context.
          * @param part Local partition.
-         * @param cpFlag Checkpoint flag.
+         * @param idleCheck Idle check operation.
          */
         private Future<Map<PartitionKeyV2, PartitionHashRecordV2>> calculatePartitionHashAsync(
             final CacheGroupContext grpCtx,
             final GridDhtLocalPartition part,
-            AtomicBoolean cpFlag
+            Runnable idleCheck
         ) {
-            return ForkJoinPool.commonPool().submit(() -> calculatePartitionHash(grpCtx, part, cpFlag));
+            return ForkJoinPool.commonPool().submit(() -> calculatePartitionHash(grpCtx, part, idleCheck));
         }
 
         /**
          * @param grpCtx Group context.
          * @param part Local partition.
-         * @param cpFlag Checkpoint flag.
+         * @param idleCheck Idle check operation.
          */
         private Map<PartitionKeyV2, PartitionHashRecordV2> calculatePartitionHash(
             CacheGroupContext grpCtx,
             GridDhtLocalPartition part,
-            AtomicBoolean cpFlag
+            Runnable idleCheck
         ) {
             if (!part.reserve())
                 return emptyMap();
 
+            idleCheck.run();
```

Comment:
Simple moving this check to the beginning of the method may break logic of complitionCntr. I think complitionCntr increment should be moved to finally block like in validate_indexes